lib/phobos_db_checkpoint/failure.rb

```diff
@@ -0,0 +1,49 @@
+module PhobosDBCheckpoint
+  class Failure < ActiveRecord::Base
+    include PhobosDBCheckpoint::EventHelper
+
+    scope :by_checksum, -> (val) { where("metadata->>'checksum' = ?", val) }
+    scope :by_topic, -> (val) { where("metadata->>'topic' = ?", val) }
```

Comment:
We chat offline already, so just bringing the discussion online.

We think we cannot use the json queries, since it's not supported by all DBs. Therefore we concluded it's better to create columns and indexes for `topic` and `group_id` so that we can filter failures later on.